packages/components/src/DataList/components/DataListHeaderTile/components/DataListSortingOptions.tsx

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import { useOnKeyDown } from "@jobber/hooks/useOnKeyDown";
+import { useRefocusOnActivator } from "@jobber/hooks/useRefocusOnActivator";
+import { Icon } from "@jobber/components/Icon";
+import styles from "./DataListSortingOptions.css";
+import { SortableOptions } from "../../../DataList.types";
+
+interface DataListSortingOptionsProps {
+  readonly options: SortableOptions[];
+  readonly selectedOption: SortableOptions | null;
+  readonly onSelectChange: (selectedOption: SortableOptions) => void;
+  readonly onClose: () => void;
+  readonly optionsListRef: React.RefObject<HTMLUListElement>;
+  readonly dataListHeaderTileRef: React.RefObject<HTMLElement>;
+}
+
+export function DataListSortingOptions({
```

Comment:
Just wondering if there's a reason this couldn't just be a generic dropdown for re-use? Most of this is what you'd get in a headless dropdown and could then be used across all components that need dropdown functionality.

I'm including this as an example, not as a suggestion, but Radix gives this all as primitives that you then style/configure to the needs. https://www.radix-ui.com/primitives/docs/components/dropdown-menu

I worry that every time we need the same functionality that looks slightly differently, we're re-implementing the behaviour.

Reply:
Hey @ericchernuka ! This is a great talking point and question.  Apologies in advance for a bit of a long winded response but I feel that some context is relevant.

Part of our discovery in September when building `Combobox` was to look at our existing selection type components (`MultiSelect`, `Select`, `Menu` etc) and evaluate if any of those could be used to build up our `ComboboxContent` - we have the components already so why not? However, the look and functionality that `Combobox` needed, meant the best path forward was to build our own dropdown within Combobox.  

In quick succession we then were faced with building an actions dropdown in `DataList` and now this sorting dropdown - where yes, you have a 'trigger' with a dropdown menu (the purpose of `Combobox`).  The snag there is that a `Combobox` trigger is either an Atlantis `Chip` or `Button`.  Given that `DataListSortingOptions` is triggered by `Text`, right now we cannot use `Combobox`.  

tldr: Yes, Atlantis needs to evaluate our current selection components, gather each of their requirements and then build one reusable dropdown. Unfortunately that is not something we have capacity for immediately however it's on the horizon.